packages/accounts-password/password_server.js

```diff
@@ -573,119 +573,187 @@ Accounts.sendResetPasswordEmail = function (userId, email) {
+  }});
 
+  // before passing to template, update user object with new token
+  Meteor._ensure(user, 'services', 'email');
+  if (!user.services.email.verificationTokens) {
+    user.services.email.verificationTokens = [];
+  }
+  user.services.email.verificationTokens.push(tokenRecord);
+};
+
+/**
+ * @summary Creates options for email sending for reset password and enroll account emails.
+ * You can use this function when customizing a reset password or enroll account email sending.
+ * @locus Server
+ * @param {Object} email Which address of the user's to send the email to.
+ * @param {Object} user The user object to generate options for.
+ * @param {String} url URL to which user is directed to confirm the email.
+ * @param {String} reason `resetPassword` or `enrollAccount`.
+ * @returns {Object} Options which can be passed to `Email.send`.
+ * @importFromPackage accounts-base
+ */
+Accounts.generateOptionsForEmail = function (email, user, url, reason) {
   var options = {
     to: email,
-    from: Accounts.emailTemplates.enrollAccount.from
-      ? Accounts.emailTemplates.enrollAccount.from(user)
+    from: Accounts.emailTemplates[reason].from
+      ? Accounts.emailTemplates[reason].from(user)
       : Accounts.emailTemplates.from,
-    subject: Accounts.emailTemplates.enrollAccount.subject(user)
+    subject: Accounts.emailTemplates[reason].subject(user)
   };
 
-  if (typeof Accounts.emailTemplates.enrollAccount.text === 'function') {
-    options.text =
-      Accounts.emailTemplates.enrollAccount.text(user, enrollAccountUrl);
+  if (typeof Accounts.emailTemplates[reason].text === 'function') {
+    options.text = Accounts.emailTemplates[reason].text(user, url);
   }
 
-  if (typeof Accounts.emailTemplates.enrollAccount.html === 'function') {
-    options.html =
-      Accounts.emailTemplates.enrollAccount.html(user, enrollAccountUrl);
+  if (typeof Accounts.emailTemplates[reason].html === 'function') {
+    options.html = Accounts.emailTemplates[reason].html(user, url);
   }
 
   if (typeof Accounts.emailTemplates.headers === 'object') {
     options.headers = Accounts.emailTemplates.headers;
   }
 
+  return options;
+};
+
+// send the user an email with a link that when opened allows the user
+// to set a new password, without the old password.
+
+/**
+ * @summary Send an email with a link the user can use to reset their password.
+ * @locus Server
+ * @param {String} userId The id of the user to send email to.
+ * @param {String} [email] Optional. Which address of the user's to send the email to. This address must be in the user's `emails` list. Defaults to the first email in the list.
+ * @returns {Object} Object with {email, user, tokenRecord, url, options} values.
+ * @importFromPackage accounts-base
+ */
+Accounts.sendResetPasswordEmail = function (userId, email, extraTokenData) {
+  const {email: realEmail, user, tokenRecord} = Accounts.generateResetTokenRecord(userId, email, 'resetPassword', extraTokenData);
```

Comment:
Ehm, comment from @abernix was that I do not have to do that: https://github.com/meteor/meteor/pull/8207#discussion_r94324213

At least this is how I understood his comment. :-)